04.ls/ls.rb

```diff
@@ -0,0 +1,29 @@
+#! /usr/bin/env ruby
+# frozen_string_literal: true
+
+# 列数を指定する
+COLUMN_SIZE = 3
+# 文字幅調整用のマージン
+PADDING_MARGIN = 2
+
+children = Dir.children(__dir__)
+# 大文字の昇順 > 小文字の昇順に並び替える
+#   ls の挙動に合わせて .から始まるファイルを除外する
+children.sort!.reject! { |child| child.start_with?('.') }
+
+# 文字幅調整用に最大文字数を取得する
+padding_size = children.map(&:length).max + PADDING_MARGIN
+
+# 配列がCOMUNS_SIZEの倍数になるように空文字で埋める
+lack_count = children.length % COLUMN_SIZE
+children.concat([''] * (COLUMN_SIZE - lack_count)) if lack_count.positive?
+
+# 行列を入れ替えた二次元配列を作成する
+divider = (children.length / COLUMN_SIZE)
+matrix = children.each_slice(divider).to_a.transpose
+matrix.each do |row|
+  row.each do |item|
```

Comment:
12行目ではchildって呼んでますよね？
同じデータは一貫して同じ名前を付ける方がいいかなと思います。
（childにするかitemにするかは要検討）

Reply:
childrenという名前は重要ではないのでitemsに差し替えようと思います

Reply:
> childrenという名前は重要ではないのでitemsに差し替えようと思います

よいと思います 👍 

ついでに、rowもitemsにした方が可読性の面でややベターな気もします。

```suggestion
  items.each do |item|
```

https://bootcamp.fjord.jp/pages/readable-variable-name#%E3%83%AB%E3%83%BC%E3%83%97%E5%87%A6%E7%90%86%E3%81%AE%E5%A0%B4%E5%90%88%E3%81%AF%E3%80%8C%E9%85%8D%E5%88%97%E3%81%AF%E8%A4%87%E6%95%B0%E5%BD%A2%E3%80%81%E3%81%9D%E3%81%AE%E4%B8%AD%E3%81%AE%E5%90%84%E8%A6%81%E7%B4%A0%E3%81%AF%E5%8D%98%E6%95%B0%E5%BD%A2%E3%80%8D%E3%81%8C%E3%82%BB%E3%82%AA%E3%83%AA%E3%83%BC